server/util/fspath/fspath.go

```diff
@@ -0,0 +1,47 @@
+package fspath
+
+import (
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+)
+
+// Whether the filesystems most commonly used on the current platform are
+// case-insensitive.
+//
+// TODO: actually check the filesystem - it's possible (but uncommon) to have
```

Comment:
I guess we could just create and open a tmp file in an `init` function. Maybe we do this now to avoid ugly bugs if our assumption here doesn't apply?

Also note that per https://learn.microsoft.com/en-us/windows/wsl/case-sensitivity you can actually toggle case sensitivity per directory. If we can, we should probably test our build root instead of an arbitrary tmp path.

Reply:
Done - testing the build root directory now.